app/javascript/mastodon/actions/statuses.js

```diff
@@ -188,6 +191,42 @@ export function deleteStatusFail(id, error) {
   };
 };
 
+export function hide(status) {
+  return function (dispatch, getState) {
+    dispatch(hideRequest(status));
+
+    api(getState).post(`/api/v1/statuses/${status.get('id')}/hide`).then(function (response) {
+      dispatch(importFetchedStatus(response.data));
+      dispatch(hideSuccess(status.get('id'), getState().get('statuses')));
```

Comment:
I'm not sure why you'd pass the entire list of statuses in that event.
Looking at the reducer code, you probably want to get the boosts instead (see deleteFromTimelines in app/javascript/mastodon/actions/timelines.js)